runtime/app/app.go

```diff
@@ -94,3 +95,13 @@ func ModusHomeDir() string {
 	}
 	return modusHome
 }
+
+func KubernetesNamespace() string {
+	if ns := os.Getenv("NAMESPACE"); ns != "" {
+		return ns
+	}
+	if data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		return strings.TrimSpace(string(data))
+	}
+	return "default"
```

Comment:
should we return an error/panic? i think it's safe to assume that if someone enables the Kubernetes cluster mode, then at least one of the two code paths above should succeed.